### PR TITLE
Proposed C5 conventions for configuring rubocop-rspec

### PR DESCRIPTION
**Pull request by mattbrictson**
I'm using rubocop-rspec a current project and have been pleased with the results. This PR ports my team's learnings back to the C5 conventions so that they can hopefully benefit future C5 Rails projects.

In general the out-of-the-box rubocop-rspec rules lead to more consistent, easier to read specs. There are a few rules that I've relaxed a bit for convenience.

The only significant departure from the defaults is for request/system specs. Our style of writing system specs is to mimic how a user would interact with the app, which usually involves taking several steps: clicking links, filling out form fields, etc. This isn't really compatible with RuboCop's preference for extremely short examples.

### CODE REVIEW

**Comment by mattbrictson** [comment]
I've broken this out as a separate file so that projects can opt-in. If I were to modify `.rubocop.yml` this would break downstream projects as they likely do not have the requisite `rubocop-rspec` gem installed.

**Comment by bunnymatic** [comment]
I'll try this on my current project and see how it flies.  the preference for `it { is_expected_to }` worries me a bit as we've been juggling back and forth between really using `let` and `subject` vs having each `it` block have self contained setup.  For new developers the latter is much easier to reason about and read.  But I'll see what `rubocop` says about what we have and maybe that'll inform.

**Comment by mattbrictson** [comment]
My understanding is that the `RSpec/ImplicitSubject` cop only applies to specs without descriptions (i.e. you are already relying on RSpec to magically describe the spec). So e.g. this would get flagged:

```ruby
it do
  expect(subject).to be_truthy
end
```

And this would get flagged:

```ruby
it { should be_truthy }
```

In both cases RuboCop would suggest:

```ruby
it { is_expected.to be_truthy }
```

But a more explicit example with a description would be fine as-is:

```ruby
it "is truthy" do
  expect(subject).to be_truthy
end
```

**Comment by mattbrictson** [comment]
tldr I don't think `RSpec/ImplicitSubject` will actually give you any problems. 🙂

**Comment by christiannelson** [comment]
I think this is a great first step and I'm excited to give it a try on some of our smaller internal projects.

File: rubocop/rubocop_rspec.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# syntax instead of `expect(subject)`, like this:
+# it { is_expected.to be_truthy }
+RSpec/ImplicitSubject:
+  EnforcedStyle: single_statement_only
+
+# Ideal BDD is to have one `expect` per example (so that each specification can
+# have corresponding documentation in the form of an `it` block), but we allow
+# some flexibility (up to 3).
+RSpec/MultipleExpectations:
+  Max: 3 # Default is 1
```
Comment (bunnymatic) [inline rubocop/rubocop_rspec.yml:33]: Because system tests are often long, the also often have many expectations.
Perhaps:
```suggestion
  Max: 3 # Default is 1
  Exclude:
    - "spec/requests/**/*"
    - "spec/system/**/*"

```

**Review by bunnymatic — APPROVED** [review]

File: rubocop/rubocop_rspec.yml (hunk 1 above)
Comment (mattbrictson) [reply rubocop/rubocop_rspec.yml:33]: Makes sense 👍

Comment (mattbrictson) [reply rubocop/rubocop_rspec.yml:33]: Would you be OK with excluding only `system` and not `requests`? In my experience request specs are used for APIs and those are easier to write bite-sized. Whereas system specs are long capybara flows.

Comment (christiannelson) [reply rubocop/rubocop_rspec.yml:33]: Hmmmm, I put capybara "system" specs under `spec/features/`.

Comment (bunnymatic) [reply rubocop/rubocop_rspec.yml:33]: leaving out request specs seems fine.  so `spec/features/**` and `spec/system/**` ?

Comment (bunnymatic) [reply rubocop/rubocop_rspec.yml:33]: i see that's done!

**Review by bunnymatic — APPROVED** [review]

**Comment by mattbrictson** [comment]
@christiannelson good to merge?

**Review by christiannelson — APPROVED** [review]
LGTM! We may have additional feedback as folks use it, but that's 100% ok. :)
